src/vivarium_census_prl_synth_pop/components/person.py

```diff
@@ -253,25 +185,47 @@ def _perform_gq_person_moves(self, pop: pd.DataFrame, movers: pd.Index) -> pd.Da
 
         return pop
 
-    def _get_new_household_ids(self, pop: pd.DataFrame, sims_who_move: pd.Index) -> pd.Series:
-        households = pop["household_id"]
-        all_household_ids = list(
-            households.squeeze().drop_duplicates()
-        )  # all household_ids in simulation
+    def _perform_non_reference_person_moves(
+        self, pop: pd.DataFrame, movers: pd.Index
+    ) -> pd.DataFrame:
+        """
+        Move each simulant in movers to a random (different) non-GQ household
+        with relationship "Other nonrelative".
+        """
+        if len(movers) == 0:
+            return pop
 
-        new_household_ids = pop.loc[
-            sims_who_move, "household_id"
-        ].copy()  # People who move household_ids
+        non_gq_address_ids = self._get_address_ids_by_household_id(
+            pop[pop["housing_type"] == "Standard"]
+        )
 
+        # NOTE: Unlike in GQ person moves, we require that a "move"
+        # not start and end in the same household.
+        to_move = movers
+        household_id_values = pop.loc[movers, "household_id"]
         additional_seed = 0
-        while (new_household_ids == pop.loc[sims_who_move, "household_id"]).any():
-            unchanged_households = new_household_ids == pop.loc[sims_who_move, "household_id"]
-            new_household_ids[unchanged_households] = self.randomness.choice(
-                new_household_ids.loc[unchanged_households].index,
-                all_household_ids,
-                additional_key=f"newHousehold_ids_{additional_seed}",
+
+        while not to_move.empty:
+            household_id_values[to_move] = self.randomness.choice(
+                to_move,
+                choices=list(non_gq_address_ids.index),
+                additional_key=f"non_reference_person_move_household_ids_{additional_seed}",
             )
-            # Some sims can move to the same house from randomness
+            to_move = movers[household_id_values == pop.loc[movers, "household_id"]]
             additional_seed += 1
 
-        return pd.Series(new_household_ids)
+        pop.loc[movers, "household_id"] = household_id_values
+        pop.loc[movers, "housing_type"] = "Standard"
+        pop.loc[movers, "address_id"] = household_id_values.map(non_gq_address_ids)
+        pop.loc[movers, "relation_to_household_head"] = "Other nonrelative"
+
+        return pop
+
+    def _get_address_ids_by_household_id(self, df: pd.DataFrame) -> pd.Series:
+        result = (
+            df[["household_id", "address_id"]]
+            .drop_duplicates()
+            .set_index("household_id")["address_id"]
+        )
+        assert result.index.is_unique
```

Comment:
No need for this assert. You have `drop_duplicates`. If we care enough about this to test it, we should write a test for it.

Reply:
`drop_duplicates` does not ensure per-column uniqueness (in this case "household_id") because it only drops duplicate _combinations_.

It makes sense why having an `assert` here is not right, but note that this component will error anyway (when it tries to use this mapping) if the index is not unique.